lib/jwt.js

```diff
@@ -0,0 +1,173 @@
+var urlDecodeB64 = function(data) {
+  return new Buffer(data, 'base64').toString('ascii');
+}
+
+var isNumber = function(n) { return typeof n === 'number' };
+
+/**
+ * Decodes a string token into the 3 parts, throws if the format is invalid
+ * @param token
+ */
+var decode = function(token) {
+  var parts = token.split('.');
+
+  if (parts.length !== 3) {
+    throw new Error('ID token could not be decoded')
+  }
+
+  var header = parts[0];
+  var payload = parts[1];
+  var signature = parts[2];
+
+  return {
+    _raw: token,
+    header: JSON.parse(urlDecodeB64(header)),
+    payload: JSON.parse(urlDecodeB64(payload)),
+    signature: signature
+  };
+}
+
+/**
+ * Verifies an idToken is valid and matches the options passed in
+ * @param idToken
+ * @param options
+ */
+var verify = function(idToken, options) {
+  if (!idToken) {
+    throw new Error('ID token is required but missing');
+  }
+
+  var decoded = decode(idToken);
```

Comment:
I avoided destructuring in this PR because none has been present in the library.  Support was added for it in Node 6.4.0, which anything before is EOL, but I didn't think it was necessary.